app/controllers/units_controller.rb

```diff
@@ -0,0 +1,35 @@
+class UnitsController < ApplicationController
+  def create
+    @system = System.find(params[:unit][:system_id])
```

Comment:
yeah - it really shouldn't result in a 500. you should gracefully handle for the absence of a `params[:unit]`